### PR TITLE
Core data experiment

### PR DESCRIPTION
**Pull request by serhii-prikhodko**
Implemented work with data via CoreData framework.

### CODE REVIEW

File: expences-app-swiftui/Core Data/CoreDataStack.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
+//
+//  CoreDataStack.swift
+//  JournalCoreDataSwiftUI
+//
+//  Created by Nelson Gonzalez on 12/10/19.
```
Comment (yaroslavluchyt) [inline expences-app-swiftui/Core Data/CoreDataStack.swift:5]: you maybe should change this

File: expences-app-swiftui/Model/ExpensesStore.swift [hunk 2]
```diff
@@ -4,48 +4,99 @@
...
+    init() {
+        getExpenses()
+    }
     
-    static func loadExpenses() -> [ExpensesByPerson] {
-        let data = JSONParser.parse(from: "expenses")
-        let expensesData = data.expenses
+     
+    func saveContext() {
+        let moc = CoreDataStack.shared.mainContext
```
Comment (yaroslavluchyt) [inline expences-app-swiftui/Model/ExpensesStore.swift:21]: `moc` isn't the best name here, please use `mainContext` or something similar

File: expences-app-swiftui/Model/ExpensesStore.swift [hunk 3]
```diff
@@ -4,48 +4,99 @@
...
+    func updateExpense(expense: ExpenseItem, name: String, amount: Double) {
+        expense.name = name
+        expense.amount = amount
+        
+        saveContext()
     }
     
     func deleteExpense(personIndex: Int, dayIndex: Int, at offsets: IndexSet) {
-        expenses[personIndex].weeklyExpenses[dayIndex].dailyExpenses.remove(atOffsets: offsets)
+        let moc = CoreDataStack.shared.mainContext
```
Comment (yaroslavluchyt) [inline expences-app-swiftui/Model/ExpensesStore.swift:59]: same as above

File: expences-app-swiftui/Model/ExpensesStore.swift [hunk 4]
```diff
@@ -4,48 +4,99 @@
...
+        do {
+            expenses = try moc.fetch(fetchRequest)
+        } catch {
+            NSLog("Error fetching expenses: \(error)")
+        }
     }
     
-    func addExpense(personIndex: Int, dayIndex: Int, expense: Expense) {
-        expenses[personIndex].weeklyExpenses[dayIndex].dailyExpenses.append(expense)
+    // MARK: CRUD methods for expense
```
Comment (yaroslavluchyt) [inline expences-app-swiftui/Model/ExpensesStore.swift:42]: nit: add a single vertical space here

File: expences-app-swiftui/Model/ExpensesStore.swift [hunk 5]
```diff
@@ -4,48 +4,99 @@
...
+        
+        saveContext()
     }
     
+    // MARK: CRUD methods for person
     func addPerson(name: String) {
-        let weeklyExpenses = [DailyExpenses.emptyDailyExpenses(), DailyExpenses.emptyDailyExpenses(), DailyExpenses.emptyDailyExpenses()]
-        let expensesByPerson = ExpensesByPerson(name: name, weeklyExpenses: weeklyExpenses)
-        expenses.append(expensesByPerson)
+        let moc = CoreDataStack.shared.mainContext
```
Comment (yaroslavluchyt) [inline expences-app-swiftui/Model/ExpensesStore.swift:71]: same as above

File: expences-app-swiftui/Model/ExpensesStore.swift [hunk 6]
```diff
@@ -4,48 +4,99 @@
...
+        
+        for index in offsets {
+            let expense = expenses[personIndex].getDailyExpensesByIndex(index: dayIndex).expensesArray[index]
+            moc.delete(expense)
+        }
+        
+        saveContext()
     }
     
+    // MARK: CRUD methods for person
```
Comment (yaroslavluchyt) [inline expences-app-swiftui/Model/ExpensesStore.swift:69]: add space here too

**Review by yaroslavluchyt — COMMENTED** [review]
looks good
let me know when you address the comments

**Comment by serhii-prikhodko** [comment]
Fixed all comments

**Review by yaroslavluchyt — APPROVED** [review]

**Pull request merged**